client/src/store/Filetypes.ts

```diff
@@ -0,0 +1,48 @@
+import { Module } from 'vuex';
+import { getValidFileTypes } from '@/lib/api/viame.service';
+
+export interface FiletypeState {
+    filetypes: null | Record<string, string[]>;
```

Comment:
You actually could more strictly type this, since you know the `filetypes` object will consist of `image`, `video` and `web` at the top level. I don't think it's necessary though.